src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -392,7 +392,7 @@ private static mdToken FindGenericMethodArgTypeSpec(EcmaModule module)
 
             // Should be unreachable - couldn't find a TypeSpec.
             // Are we still compiling CoreLib?
-            throw new NotSupportedException();
+            throw new UnreachableException();
```

Comment:
ILCompiler and other related is not compiled against current netcoreapp. It may be best to revert this for it.